static/js/components/tiles/histogram_tile.tsx

```diff
@@ -18,177 +18,202 @@
-  }, [props]);
-
-  useEffect(() => {
-    if (rawData) {
-      processData(props, rawData, setHistogramData);
-    }
-  }, [props, rawData]);
+  // specify full dates if start and end dates are just YYYY
+  if (startDate.length == "YYYY".length && endDate.length == "YYYY".length) {
+    startDate = `${startDate}-01-01`;
+    endDate = `${endDate}-12-31`;
+  }
 
-  useEffect(() => {
-    if (histogramData) {
-      draw(props, histogramData.dataPoints);
+  // Update end date to the latest date we have data for, if our data ends
+  // before the end of the date range. This prevents us from incorrectly
+  // imputing 0s for dates the source(s) may not yet have data for.
+  if (!_.isEmpty(disasterEventPoints)) {
+    // Sort events into chronological order
+    disasterEventPoints.sort((a, b) =>
+      a.startDate < b.startDate ? -1 : a.startDate > b.startDate ? 1 : 0
+    );
+    const lastDate =
+      disasterEventPoints[disasterEventPoints.length - 1].startDate;
+    if (lastDate < endDate) {
+      endDate = lastDate;
     }
-  }, [props, histogramData]);
-
-  if (!histogramData) {
-    return null;
   }
-  const rs: ReplacementStrings = {
-    place: props.place.name,
-    date: "",
-  };
-  return (
-    <ChartTileContainer
-      title={props.title}
-      sources={histogramData.sources}
-      replacementStrings={rs}
-      className={`${props.className} histogram-chart`}
-      allowEmbed={true}
-      getDataCsv={() => dataPointsToCsv(histogramData.dataPoints)}
-    >
-      <div id={props.id} className="svg-container" ref={svgContainer}></div>
-    </ChartTileContainer>
-  );
+  // Fill in months between start and end dates
+  const months = new Array<string>();
+  for (
+    let date = new Date(startDate + "Z");
+    getMonthString(date) <= getMonthString(new Date(endDate + "Z"));
+    date.setMonth(date.getMonth() + 1)
+  ) {
+    months.push(getMonthString(new Date(date)));
+  }
+  return months;
 }
 
 /**
- * Gets the series of observations to plot
+ * Helper function to bin data by month
  */
-function fetchData(
-  props: HistogramTilePropType,
-  setRawData: (data: SeriesApiResponse) => void
-): void {
-  const statVars = [];
-  for (const spec of props.statVarSpec) {
-    statVars.push(spec.statVar);
-    if (spec.denom) {
-      statVars.push(spec.denom);
+function binDataByMonth(
+  disasterEventPoints: DisasterEventPoint[],
+  dateSetting: string
+): DataPoint[] {
+  if (_.isEmpty(disasterEventPoints)) {
+    return [];
+  }
+
+  // Track YYYY-MM -> number of events
+  const bins = new Map<string, number>();
+
+  // Initialize all bins at zero
+  const monthsToPlot = getMonthsArray(dateSetting, disasterEventPoints);
+  for (const month of monthsToPlot) {
+    bins.set(month, 0);
+  }
+
+  for (const event of disasterEventPoints) {
+    // Get start time in YYYY-MM
+    const eventMonth = event.startDate.slice(0, "YYYY-MM".length);
+
+    // Increment count in corresponding bin
+    if (bins.has(eventMonth)) {
```

Comment:
perhaps add a log warning if we're skipping events.

Reply:
added a log message if we skip events